File: array/medium/minimize_the_heights_2/code_1.py
class Solution:
    def getMinDiff(self, arr,k):
        # code here
        arr.sort()
        
        ans = arr[-1] - arr[0]
        
        for i in range(1, len(arr)):
            if arr[i] - k < 0:
                continue
            
            minElem = min(arr[0] + k, arr[i] - k)
            maxElem = max(arr[i - 1] + k, arr[-1] - k)
            
            ans = min(ans, maxElem - minElem)
        
        return ans

File: array/medium/maximum_product_subarray/code_1.c
int max(int a, int b){
    return a > b ? a : b;
}

int min(int a, int b){
    return a < b? a : b;
}

// Function to find maximum product subarray
int maxProduct(int arr[], int n) {
    // code here...
    int max_current = arr[0];
    int min_current = arr[0];
    int req_max = arr[0];
    
    for(int i = 1; i < n; i++){
        int temp = max_current;
        
        max_current = max(arr[i], max(arr[i] * max_current, arr[i] * min_current));
        min_current = min(arr[i], min(arr[i] * temp, arr[i] * min_current));
        
        req_max = max(req_max, max_current);
    }
    
    return req_max;
}